### ISSUE TITLE
Show complete include hierarchy

### ISSUE THREAD

**Issue by FStefanni** [clangd/vscode-clangd#416]
Hi,

not sure if this is where to report this feature request.
Basically, when there is an error into an included file, the message reports something like this:
```
myheader.hpp
/path/include/coreino.hpp

In included file: <error description>
otherHeader.h(579, 5): Error occurred here
No quick fixes available
```

This make the error not so trivial to fix, since it only report the header name where the error originates, but not all the includes hierarchy to arrive to that include file.

For example, this would be more useful:
```
myheader.hpp
/path/include/coreino.hpp

In included file: <error description>
otherHeader.h(579, 5): Error occurred here
included by: another0.h(23,5)
included by: another1.h(11,4)
included by: another3.h(57,9)
myheader.hpp(2,1)
No quick fixes available
```

This is especially useful in cases where we are maybe cross-compiling, or in complex setups where we have to specify to clangd custom include paths, and the error is basically that clangd comes up to see as included the wrong file (i.e., in the example exist two different otherHeader.h  files, and the one causing the error is the wrong one).

Regards

**Comment by HighCommander4** [comment]
Agreed, this is important context for understanding a diagnostic. I would expect clangd to produce a "note" of the form "In file included from <filename>" for every element of the include chain, similar to how the compiler would display the diagnostic on the command line.
